### PR TITLE
[Feature] Support month, day, and hour expressions for cdc synchroniz…

### LINKED ISSUE #1141: [Feature] Support month, day, and hour expressions for cdc synchronizing tables action
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Currently, cdc synchronizing tables action support computed column. We can support more expressions:
month: extract month from date column
day：extract day from date column
hour：extract hour from date or time column

### Solution

Please reference the implementation of year:
#1109

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by pizhihui**
### Purpose
Support month, day, and hour expressions for cdc synchronizing tables action

Linked issue: close #1141 

### Tests

`MySqlSyncTableActionITCase`

### API and Format

New argument in mysql-sync-table:
--computed-column

add month,day and hour expression for `--computed-column` params

### Documentation

CDC Ingestion

### CODE REVIEW

File: docs/layouts/shortcodes/generated/kafka_sync_table.html [hunk 1]
```diff
@@ -47,7 +47,7 @@
...
...om a DATE, DATETIME or TIMESTAMP. Output is an INT value represent the month of year.</li><li>day(date-column): Extract day of month from a DATE, DATETIME or TIMESTAMP. Output is an INT value represent the day of month.</li><li>hour(date-column): Extract hour from a DATE, DATETIME or TIMESTAMP. Output is an INT value represent the hour.</li><li>substring(column,beginInclusive): Get column.substring(beginInclusive). Output is a STRING.</li><li>substring(column,beginInclusive,endExclusive): Get column.substring(beginInclusive,endExclusive). Output is a STRING.</li><li>truncate(column,width): truncate column by width. Output type is same with column.<ul><li>If the column is a STRING, truncate(column,width) will truncate the string to width characters, namely "value.substring(0, width)".</li><li>If the column is an INT or LONG, truncate(column,width) will truncate the number with the algorithm "v - (((v % W) + W) % W)". The "redundant" compute part is to keep the result always positive.</li><li>If the column is a DECIMAL, truncate(column,width) will truncate the decimal with the algorithm: let "scaled_W = decimal(W, scale(v))", then return "v - (v % scaled_W)".</li></ul></li></ul></td>
```
Comment (JingsongLi) [inline docs/layouts/shortcodes/generated/kafka_sync_table.html:50]: Remove `Supported expressions are.......`, It duplicates the content below the document

File: docs/layouts/shortcodes/generated/mysql_sync_table.html [hunk 2]
```diff
@@ -47,7 +47,7 @@
...
...om a DATE, DATETIME or TIMESTAMP. Output is an INT value represent the month of year.</li><li>day(date-column): Extract day of month from a DATE, DATETIME or TIMESTAMP. Output is an INT value represent the day of month.</li><li>hour(date-column): Extract hour from a DATE, DATETIME or TIMESTAMP. Output is an INT value represent the hour.</li><li>substring(column,beginInclusive): Get column.substring(beginInclusive). Output is a STRING.</li><li>substring(column,beginInclusive,endExclusive): Get column.substring(beginInclusive,endExclusive). Output is a STRING.</li><li>truncate(column,width): truncate column by width. Output type is same with column.<ul><li>If the column is a STRING, truncate(column,width) will truncate the string to width characters, namely "value.substring(0, width)".</li><li>If the column is an INT or LONG, truncate(column,width) will truncate the number with the algorithm "v - (((v % W) + W) % W)". The "redundant" compute part is to keep the result always positive.</li><li>If the column is a DECIMAL, truncate(column,width) will truncate the decimal with the algorithm: let "scaled_W = decimal(W, scale(v))", then return "v - (v % scaled_W)".</li></ul></li></ul></td>
```
Comment (JingsongLi) [inline docs/layouts/shortcodes/generated/mysql_sync_table.html:50]: Remove `Supported expressions are.......`, It duplicates the content below the document

**Review by JingsongLi — COMMENTED** [review]
Thanks @pizhihui for the contribution, left comments.

File: docs/layouts/shortcodes/generated/kafka_sync_table.html (hunk 1 above)
Comment (pizhihui) [reply docs/layouts/shortcodes/generated/kafka_sync_table.html]: I remove it, and add a anchor `See here ......`  point to jump to the computed functions item.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/Expression.java [hunk 3]
```diff
@@ -136,6 +155,90 @@ public String eval(String input) {
...
+        }
+
+        @Override
+        public DataType outputType() {
+            return DataTypes.INT();
+        }
+
+        @Override
+        public String eval(String input) {
+            LocalDateTime localDateTime = DateTimeUtils.toLocalDateTime(input, 0);
```
[the remark this answers is not in the document]
Comment (pizhihui) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/Expression.java]: I added a method with one parameter to the `DateTimeUtils` class

[the remark this answers is not in the document]
Comment (JingsongLi) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/Expression.java]: Sorry, I did not mean to extract `DateTimeUtils.toLocalDateTime(input)`, just keep previous code is OK.

[the remark this answers is not in the document]
Comment (JingsongLi) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/Expression.java]: You can take a look to Flink `DateTimeUtils.parseTimestampData`. The default value is 3.

[the remark this answers is not in the document]
Comment (pizhihui) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/Expression.java:181]: 1. Thanks for your hint,I used previous code.
2. Flink `DateTimeUtils.parseTimestampData` is linked to FLINK-14925, What I understand is that it needs to be consistent with Flink,So what's the best we can do.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**